media/starboard/starboard_renderer.cc

```diff
@@ -127,25 +127,42 @@ void StarboardRenderer::Initialize(MediaResource* media_resource,
     return;
   }
 
-#if COBALT_MEDIA_ENABLE_ENCRYPTED_PLAYBACKS
-  base::AutoLock auto_lock(lock_);
+  // TODO(sideboard) ensure SetCdm has been called before |CreatePlayerBridge()|
+  // and the init_cb.
+  //
+  // base::AutoLock auto_lock(lock_); bool is_encrypted =
+  //     audio_stream_ && audio_stream_->audio_decoder_config().is_encrypted();
+  // is_encrypted |=
+  //     video_stream_ && video_stream_->video_decoder_config().is_encrypted();
+  // if (is_encrypted) {
+  //   // TODO(b/328305808): Shall we call client_->OnVideoNaturalSizeChange()
+  //   to
+  //   // provide an initial size before the license exchange finishes?
+
+  //   RunSetDrmSystemReadyCB(BindPostTaskToCurrentDefault(
+  //       base::Bind(&SbPlayerPipeline::CreatePlayer, this)));
+  //   return;
+  // }
+
+  // |init_cb| will be called inside |CreatePlayerBridge()|.
+  CreatePlayerBridge(std::move(init_cb));
+}
 
-  bool is_encrypted =
-      audio_stream_ && audio_stream_->audio_decoder_config().is_encrypted();
-  is_encrypted |=
-      video_stream_ && video_stream_->video_decoder_config().is_encrypted();
-  if (is_encrypted) {
-    // TODO(b/328305808): Shall we call client_->OnVideoNaturalSizeChange() to
-    // provide an initial size before the license exchange finishes?
+void StarboardRenderer::SetCdm(CdmContext* cdm_context,
+                               CdmAttachedCB cdm_attached_cb) {
+  DCHECK(task_runner_->RunsTasksInCurrentSequence());
+  DCHECK(cdm_context);
+  TRACE_EVENT0("media", "StarboardRenderer::SetCdm");
 
-    RunSetDrmSystemReadyCB(BindPostTaskToCurrentDefault(
-        base::Bind(&SbPlayerPipeline::CreatePlayer, this)));
+  if (SbDrmSystemIsValid(drm_system_)) {
+    LOG(WARNING) << "Switching CDM not supported.";
+    std::move(cdm_attached_cb).Run(false);
     return;
   }
-#endif  // COBALT_MEDIA_ENABLE_ENCRYPTED_PLAYBACKS
 
-  // |init_cb| will be called inside |CreatePlayerBridge()|.
-  CreatePlayerBridge(std::move(init_cb));
+  drm_system_ = cdm_context->GetSbDrmSystem();
```

Comment:
If we set `use_starboard_media=false` in args.gn, this will fail to build Cobalt.apk. Consider to put the gate here, and add a TODO, where we should only build StarboardRenderer if `use_starboard_media=true`.

Reply:
done.